### ISSUE TITLE
Change `lane_number` to `lane_index`; clarify/standardize first value

### ISSUE THREAD

**Issue by j-d-b** [usdot-jpo-ode/jpo-wzdx#67]
# Change the lanes table `lane_number` field to `lane_index`; clarify/standardize first value

## Background and Issue
Currently, each lane's location is identified by an index, the `lane_number`. It is not specified what the first index should be (that is 0, 1, or 5 even--there's nothing stopping you!). One could also build the following lanes for a two-lane road and they wouldn't be violating the specification.

```json
{
  "road_event": {
    "lanes": [
      {
        "lane_number": 15
      },
      {
        "lane_number": 3
      }
    ]
  }
}
```

*note the above is pseudocode*

## Solution
Since the number is an index, the field name `lane_index` would be more clear. It should be clarified in the field description what the first/start lane index is, either 0 (I'm for this because indexes usually start at 0) or 1.

**Comment by kris-codes** [comment]
In traffic incident management, the far left (inside) lane is designated as lane 1.  This may be a topic of conversation as DOTs and developers have conversations rolling the spec out.

**Comment by sergebeaudry** [comment]
> In traffic incident management, the far left (inside) lane is designated as lane 1. This may be a topic of conversation as DOTs and developers have conversations rolling the spec out.

Currently it depends from state to state. That is why the specs is currently flexible and feed shall declare the edge reference within the lane_edge_reference field

**Comment by lynnerandolph** [comment]
As each state is going to have to implement something to convert their current representation to this representation, have the spec clearly define how it should look will reduce confusion.

**Comment by CraigMoore-Sea** [comment]
Agree this makes sense. The should be considered index values only for the specific record in question. Thus they don't have meaning in relation to another record for the same road further down the road when the lane configuration may change.

**Comment by Mahsa-Ettefagh** [comment]
Comments received during the [V3 Specification Update Subgroup](https://github.com/usdot-jpo-ode/jpo-wzdx/wiki/WZDx-v3-Specification-Update-Subgroup-Meeting-Notes,-2020-04-29) meeting on 4/29/2020:

- Hua Xiang - Maryland DOT: Which DOT use 0 as the right most lane?
- David Craig @ GM: Maps tend to start with 1 and there is no such thing as lane zero. Use lane_number
- Lynne Randolph: starting at 0 is a programming thing, a feed should start with 1
- Hua Xiang - Maryland DOT: Agree. Support: Lane Numaber, and starts with 1
- Sabrina Mosher: We currently standardize at 1 with a left edge reference in our feed. lane_index would make it confusing with lane_id
- Neil Boudreau, MassDOT: Start with 1 and use Lane Number
- Carole Delion (MDOT SHA): yes, standardized - question on whether 1 includes shoulder?
- Lynne Randolph: yes, doesn't make sense to have random numbers as indices either
- Pier Castonguay (Ver-Mac): Agreed we define lanes starting at 1 too
- Valerie Shuman: In IL, some shoulders are lanes some of the time... (e.g., for buses)
- Carole Delion (MDOT SHA): we use lane number, not index
- Skylar Knickerbocker (Iowa State): lane_number
- Jim Williams (INRIX): Lane_number dodges the "Indexes start at 0" argument.
- Chris Brookes MDOT: Is there a diagram that could be created to show this so everyone is doing it the same, once determined. Left to right or tight to left inside/out. Seems like a short example would help.
- Neil Boudreau, MassDOT: I agree with Mr. Brookes, a diagram that shows the end-user what we mean by the lane numbering configuration is the best way to ensure understanding and consistency.

**Comment by j-d-b** [comment]
Resolved in #94
